main.nf

```diff
@@ -310,74 +390,129 @@ process predict_proteins {
         }
 
     input:
-    val microbiome_id from ch_assembly_input.ids
-    file microbiome_file from ch_assembly_input.files
+    val microbiome_id from ch_nucl_input_ids
+    val bin_basename from ch_nucl_input_bin_basenames
+    file microbiome_file from ch_nucl_input_files
 
     output:
-    tuple val(microbiome_id), file("proteins.pred_${microbiome_id}.tsv.gz") into (ch_pred_proteins, ch_proteins_get_abundances)
-    file "coords.pred_${microbiome_id}.gff"
+    val microbiome_id into ch_pred_proteins_microbiome_ids                  // Emit microbiome ID
+    val bin_basename into ch_pred_proteins_bin_basename
+    file("proteins.pred_${microbiome_id}*.tsv.gz") into ch_pred_proteins     // Emit protein tsv
+    file "coords.pred_${microbiome_id}*.gff"
 
     script:
     def mode = params.prodigal_mode
+    def name = bin_basename ? "${microbiome_id}.${bin_basename}" : "${microbiome_id}"
     """
     gzip -c -d $microbiome_file | prodigal \
```

Comment:
predict_proteins

Reply:
Removed plain FASTA format from list of supported file extensions for now